features/step_definitions/cluster_configuration/localstorage.rb

```diff
@@ -1,245 +1,95 @@
-  filepath = @result[:abs_path]
-  sc["metadata"]["name"] = "local-fast"
-  sc["provisioner"] = "kubernetes.io/no-provisioner"
-  sc["volumeBindingMode"] = "Immediate"
-  File.write(filepath, sc.to_yaml)
-  @result = admin.cli_exec(:create, f: filepath)
-  step %Q/the step should succeed/
+  raise "No pod is running for local storage operator" unless @result[:success]
 
-  sc["metadata"]["name"] = "local-slow"
-  File.write(filepath, sc.to_yaml)
-  @result = admin.cli_exec(:create, f: filepath)
-  step %Q/the step should succeed/
+  logger.info(" local storage operator is installed successfully in #{namespace} namespace")
 end
 
-# local raw block devices provisioner is supported from OCP 3.10
-# Before deploy this provisioner, feature gate "BlockVolume" needs enable first.
-Given /^I deploy local raw block devices provisioner(?: with "([^ ]+?)" version)?$/ do |img_version|
+Given /^local storage provisioner has been installed successfully$/ do
   ensure_admin_tagged
-  ensure_destructive_tagged
-
-  namespace = "#{env.local_storage_provisioner_project.name}-block"
-  config_hash = env.master_services[0].config.as_hash()
-  imgformat = config_hash["imageConfig"]["format"]
-  img_registry = imgformat.split("\/")[0]
-  ose_version = env.get_version(user: admin)
-  img_version = "v#{ose_version[0]}" unless img_version
-  serviceaccount="local-storage-admin"
-  configmap="local-volume-config"
-  template="local-storage-provisioner"
-  image="#{img_registry}/openshift3/local-storage-provisioner:#{img_version}"
-  path ||="/mnt/local-storage"
-
-  project(namespace)
-  # if project exists, delete project and pvs created by local storage provisioner
-  if project.exists?(user: admin)
-    project.delete(by: admin)
-
-    BushSlicer::PersistentVolume.list(user: admin).each { |pv|
-      pv.delete(by: admin) if pv.name.start_with?("local-pv-") && pv.local_path&.start_with?("#{path}/block-devices")
-    }
-  end
-
-  ns_cmd = "oc adm new-project #{namespace} --node-selector=''"
-  @result = env.master_hosts.first.exec(ns_cmd)
-  step %Q/the step should succeed/
-
-  env.hosts.each do |host|
-    setup_commands = [
-      "losetup -d /dev/loop23 /dev/loop24",
-      "rm -rf #{path}/block-devices",
-      "mkdir -p #{path}/block-devices",
-      "dd if=/dev/zero of=#{path}/block-devices/dev1 bs=1M count=100",
-      "dd if=/dev/zero of=#{path}/block-devices/dev2 bs=1M count=100",
-      "losetup /dev/loop23 --show #{path}/block-devices/dev1",
-      "losetup /dev/loop24 --show #{path}/block-devices/dev2",
-      "ln -s /dev/loop23 #{path}/block-devices/ldev1",
-      "ln -s /dev/loop24 #{path}/block-devices/ldev2",
-      "chcon -R unconfined_u:object_r:svirt_sandbox_file_t:s0 /dev/loop23 /dev/loop24",
-      "chcon -R unconfined_u:object_r:svirt_sandbox_file_t:s0 #{path}"
-    ]
-    res = host.exec_admin(*setup_commands)
-    raise "error preaparing subdirs for local storage provisioner" unless res[:success]
+  namespace = "openshift-local-storage"
```

Comment:
@qinpingli can the local storage operator be installed in any namespace?

Reply:
Yes, it can be installed in any namespace, but we don't recommend doing this.

Reply:
ok